jetty-servlet/src/main/java/org/eclipse/jetty/servlet/ServletContextHandler.java

```diff
@@ -504,6 +505,47 @@ public FilterHolder addFilter(String filterClass, String pathSpec, EnumSet<Dispa
     {
         return getServletHandler().addFilterWithMapping(filterClass, pathSpec, dispatches);
     }
+    
+    /**
+     * Utility Method to allow for manual execution of {@link javax.servlet.ServletContainerInitializer} when using Embedded Jetty.
+     * @param containerInitializer the ServletContainerInitializer to register.
+     * @see Initializer
+     */
+    public void addServletContainerInitializer(ServletContainerInitializer containerInitializer)
+    {
+        if (!isStopped())
+            throw new IllegalStateException("ServletContainerInitializers should be added before starting");
+
+        addServletContainerInitializer(new ServletContainerInitializerHolder(containerInitializer));
+    }
+
+    /**
+     * Utility Method to allow for manual execution of {@link javax.servlet.ServletContainerInitializer} when using Embedded Jetty.
+     * @param sci the ServletContainerInitializer to register.
+     * @param classes the Set of application classes.
+     * @see Initializer
+     */
+    public void addServletContainerInitializer(ServletContainerInitializer sci, Class<?>... classes)
```

Comment:
Do we need this method, if we return the holder, onto which we can add the classes?

Reply:
I think we keep this method:  it is in the style of the other add* methods to provide option to pass in other args used by the equivalent *Holder class - look at the various addFilter methods which have args that are passed onto the FilterHolder via setters.  It think having this method signature allows users to ignore the ServletContainerInitializerHolder if they want to rather than forcing them to deal with it.

Reply:
Okay, but I still would like to see the holder be returned from the method.

Reply:
@sbordet yes, it is now.